### PR TITLE
ebos: only write out temperature in thermal runs

### PR DESCRIPTION
**Pull request by andlaus**
this PR does the same as #413 but in a more ewomsy way.

### CODE REVIEW

**Comment by andlaus** [comment]
jenkins build this please

**Comment by andlaus** [comment]
the jenkins failure should be expected: the isothermal blackoil tests don't write a TEMP field anymore after all...

**Comment by andlaus** [comment]
jenkins build this update_data please

**Comment by andlaus** [comment]
jenkins build this opm-tests=73 please

**Comment by andlaus** [comment]
BTW: the reference data in opm-tests for `flow_legacy` should be deleted. right, @atgeirr, @akva2?

File: ebos/ecloutputblackoilmodule.hh [hunk 1]
```diff
@@ -710,7 +711,7 @@ public:
             sol.insert("PRESSURE", Opm::UnitSystem::measure::pressure, std::move(oilPressure_), Opm::data::TargetType::RESTART_SOLUTION);
         }
 
-        if (temperature_.size() > 0) {
+        if (enableEnergy) {
             sol.insert("TEMP", Opm::UnitSystem::measure::temperature, std::move(temperature_), Opm::data::TargetType::RESTART_SOLUTION);
```
Comment (totto82) [inline ebos/ecloutputblackoilmodule.hh]: if (enableEnergy) should instead be added to line 232 in this file.

Comment (andlaus) [reply ebos/ecloutputblackoilmodule.hh]: thanks for noticing. changed & PR rebased.

**Comment by andlaus** [comment]
jenkins build this opm-tests=73 please

**Comment by andlaus** [comment]
hm, the jenkins error seems to be completely unreleated to this:

```
[...]
opm-simulators/opm/autodiff/BlackoilWellModel_impl.hpp:510:44: error: 'class Opm::WellTestState' has no member named 'lastTestTime'
```

 @GitPaean: you've been recently working on this. any ideas?

**Comment by andlaus** [comment]
another data update is required because opm-tests changed in the meantime:

**Comment by andlaus** [comment]
jenkins build this update_data please

**Comment by andlaus** [comment]
jenkins build this opm-tests=73 please

**Comment by andlaus** [comment]
jenkins build this update_data please

**Comment by andlaus** [comment]
jenkins build this opm-tests=73 please

**Pull request merged**